pkg/config/expand.go

```diff
@@ -429,24 +433,24 @@ func combineModuleLabels(dc *DeploymentConfig, iGrp int, iMod int) error {
 		mod.WrapSettingsWith[labels] = []string{"merge(", ")"}
 		mod.Settings[labels] = []interface{}{"((var.labels))", modLabels}
 	} else if mod.Kind == "packer" {
-		mergeInLabels(modLabels, dc.Config.Vars[labels].(map[string]interface{}))
-		mod.Settings[labels] = modLabels
+		g := map[string]interface{}{}
+		for k, v := range dc.Config.Vars.Get(labels).AsValueMap() {
+			g[k] = v.AsString()
+		}
+		mod.Settings[labels] = mergeLabels(modLabels, g)
 	}
 	return nil
 }
 
-func mergeInLabels[V interface{}](to map[string]V, from map[string]V) {
-	for k, v := range from {
-		if _, exists := to[k]; !exists {
-			to[k] = v
-		}
-	}
+func mergeLabels[V interface{}](to map[string]V, from map[string]V) map[string]V {
```

Comment:
I would add a short comment explaining the merge behavior of maps.Copy ("from" and "to" are hints, but not quite 100% of information I'd want at a glance)

Reply:
Rewrote it, added a comment.